src/SelfServe/GraphAPICompute/GraphAPICompute.tsx

```diff
@@ -0,0 +1,411 @@
+        const operationStatusUrl = await deleteComputeResource();
+        return {
+          operationStatusUrl: operationStatusUrl,
+          portalNotification: {
+            initialize: {
+              titleTKey: "DeleteInitializeTitle",
+              messageTKey: "DeleteInitializeMessage",
+            },
+            success: {
+              titleTKey: "DeleteSuccessTitle",
+              messageTKey: "DeleteSuccesseMessage",
+            },
+            failure: {
+              titleTKey: "DeleteFailureTitle",
+              messageTKey: "DeleteFailureMessage",
+            },
+          },
+        };
+      } else {
+        const sku = currentValues.get("sku")?.value as string;
+        const instances = currentValues.get("instances").value as number;
+        const operationStatusUrl = await updateComputeResource(sku, instances);
+        return {
+          operationStatusUrl: operationStatusUrl,
+          portalNotification: {
+            initialize: {
+              titleTKey: "UpdateInitializeTitle",
+              messageTKey: "UpdateInitializeMessage",
+            },
+            success: {
+              titleTKey: "UpdateSuccessTitle",
+              messageTKey: "UpdateSuccesseMessage",
+            },
+            failure: {
+              titleTKey: "UpdateFailureTitle",
+              messageTKey: "UpdateFailureMessage",
+            },
+          },
+        };
+      }
+    } else {
+      const sku = currentValues.get("sku")?.value as string;
+      const instances = currentValues.get("instances").value as number;
+      const operationStatusUrl = await updateComputeResource(sku, instances);
+      return {
+        operationStatusUrl: operationStatusUrl,
+        portalNotification: {
+          initialize: {
+            titleTKey: "CreateInitializeTitle",
+            messageTKey: "CreateInitializeMessage",
+          },
+          success: {
+            titleTKey: "CreateSuccessTitle",
+            messageTKey: "CreateSuccesseMessage",
+          },
+          failure: {
+            titleTKey: "CreateFailureTitle",
+            messageTKey: "CreateFailureMessage",
+          },
+        },
+      };
+    }
+  };
+
+  public initialize = async (): Promise<Map<string, SmartUiInput>> => {
+    // Based on the RP call enableCompute will be true if it has not yet been enabled and false if it has.
+    const defaults = new Map<string, SmartUiInput>();
+    defaults.set("enableCompute", { value: false });
+    defaults.set("sku", { value: CosmosD4s, hidden: true });
+    defaults.set("instances", { value: 1, hidden: true});
+    defaults.set("costPerHour", undefined);
+    defaults.set("connectionString", undefined);
+    defaults.set("metricsString", {
+      value: undefined,
+      hidden: true,
+    });
+
+    regions = await getReadRegions();
+    priceMap = await getPriceMap(regions);
+    const response = await getCurrentProvisioningState();
+    if (response.status && response.status !== "Deleting") {
```

Comment:
you should probably handle "Deleting", "Creating" and "Created" cases separately.
the Connection string and monitor usage sections should show up only if its "Created" and not in "Creating" state. Its ok to show it in "Deleting" state though (because that info is already shown to the user)

Reply:
Done